test/unit/specs/create-map.spec.js

```diff
@@ -61,9 +61,7 @@ describe('Creating Route Map', function () {
       '',
       '/foo',
       '/bar/',
-      '/bar',
```

Comment:
These should still work. We’re they removed by accident?

Reply:
I removed because the previous rule always match before, because of the strict flag off.

I was not convinced about this behavior, so I wrote a test to validate it.

https://github.com/buserbrasil/vue-router/commit/c247c520d121b4847d0e9d88db5368885e5bd931

I think it is an implementation detail and don't change any behavior.

It is possible to maintain them in the list, but I have to think how to do it without increasing the size.

Reply:
My change remove static rules that will never match because you have another rule matching before. This way I can trust if it exists in path map, it should be used instead of running through the matching loop.

Reply:
yeah, these need to be added back, allowing a leading slash when strict is false is necessary. If you think this is too much, that's fine, it's better to drop these changes on v3

Reply:
I changed the test to assert that the parent route is not removed from the list when child route is strict. It don't change matching behavior.

I'm ok if it is not merged and I'll work on v3 soon to avoid O(n) matches.